src/win/pipe.c

```diff
@@ -46,6 +46,9 @@ struct uv__ipc_queue_item_s {
 /* A zero-size buffer for use by uv_pipe_read */
 static char uv_zero_[] = "";
 
+/* To prevent deadlocks we will interrupt synchronous pipe read every 500 ms */
+#define PIPE_ZERO_READ_INTERRUPT_INTERVAL 500
```

Comment:
This seems very frequent - especially for something that will likely be caused by stdio, having timers that need to get woken up at a 500ms period seems like it could have a measurable impact on battery life. For non-nodejs consumers of this library (i.e. programs that don't share the nodejs bug, like my use case of JuliaLang), there should also be a way to disable this code.